### PR TITLE
supply a full version for the minimum supported openapi version

### PR DESCRIPTION
**Pull request by Leptopoda**

### CODE REVIEW

**Review by provokateurin — APPROVED** [review]
I think we should also limit the upper version. That would be 3.1 right now

**Comment by Leptopoda** [comment]
Why not limit it to the next major one (4.0.0 in this case)

**Comment by provokateurin** [comment]
Because then we are expected to also support 3.2 and other versions, but since don't claim full compatibility anyway it doesn't matter. Please pin to <4.0.0

File: packages/dynamite/dynamite/lib/src/openapi_builder.dart [hunk 1]
```diff
@@ -45,7 +48,8 @@ class OpenAPIBuilder implements Builder {
         _ => throw StateError('Openapi specs can only be yaml or json.'),
       };
 
-      if (Version.parse(spec.version).major != 3) {
+      final version = Version.parse(spec.version);
+      if (version < minSupportedVersion || version > minSupportedVersion.incrementMajor()) {
         throw Exception('Only OpenAPI 3.0.0 and later are supported');
```
Comment (provokateurin) [inline packages/dynamite/dynamite/lib/src/openapi_builder.dart:53]: Error message should also include the upper version

**Review by provokateurin — CHANGES_REQUESTED** [review]

**Review by provokateurin — APPROVED** [review]

**Pull request merged**